### ISSUE TITLE
The NuGet client ignores the NO_PROXY setting

### ISSUE THREAD

**Issue by Lukzy** [NuGet/Home#10902]
## Details about Problem

The `NO_PROXY` environment variable seems to be ignored by the NuGet client.

Looking at the source, the latest change on `WebProxy` could have caused this issue. `_bypassList` will never be set if a framework other than `NET45` is used.

https://github.com/NuGet/NuGet.Client/blob/dev/src/NuGet.Core/NuGet.Configuration/Proxy/WebProxy.cs#L61

```
        public IReadOnlyList<string> BypassList
        {
            get
            {
                return _bypassList;
            }
            set
            {
#if NET45
                _bypassList = value ?? new string[] { };
#else
                _bypassList = Array.Empty<string>();
#endif

                UpdateRegExList();
            }
        }
```

NuGet product used (NuGet.exe | Visual Studio | MSBuild.exe | dotnet.exe): **dotnet restore**

Product version: **5.10.0.7240**

Worked before? If so, with which NuGet version: **5.7.1.7**

**Comment by heng-liu** [comment]
Hi @erdembayar , could you help confirm if this [line ](https://github.com/NuGet/NuGet.Client/commit/9cb32bfc4774ae77c64ecc9ec309741f367bb444#diff-c1094245ca1788adc6314077db4d287f10da673b155cbcdedf681ac3a15eeaedR61) needs to be changed? Thanks!

**Comment by erdembayar** [comment]
@Lukzy @heng-liu 
Sorry, this is my fault. I'll commit fix today.

**Comment by erdembayar** [comment]
@Lukzy 
Can you give me repro steps? Fix is straight forward, but I need to add unit test so it wouldn't happen again. 
What sample values can I use for `http_proxy` and `no_proxy`? Currently there is no unit test actually set value in `no_proxy`.

**Comment by Lukzy** [comment]
@erdembayar
Maybe the unit tests from the dotnet runtime can help you out.

https://github.com/dotnet/runtime/blob/main/src/libraries/System.Net.WebProxy/tests/WebProxyTest.cs

I encountered the issue when I switched from the docker image `mcr.microsoft.com/dotnet/sdk:3.1-alpine` to `mcr.microsoft.com/dotnet/sdk:5.0.300-alpine3.13`.  Had a nuget.config with two package sources
```
    <packageSources>
        <add key="nuget.org" value="https://api.nuget.org/v3/index.json" protocolVersion="3" />
        <add key="TeamCity" value="http://buildserver.int/guestAuth/app/nuget/v1/FeedService.svc" />
    </packageSources>
```
And then I set the proxy over `--build-args` when running `docker build`. So this probably doesn't help you for writing unit tests.

**Comment by erdembayar** [comment]
@Lukzy @heng-liu 
Thank you for bring this bug into our attention. Fix it out: https://github.com/NuGet/NuGet.Client/pull/4079

**Comment by heng-liu** [comment]
Thanks for the super quick response!

**Comment by Starkie** [comment]
We just found this morning about the error when our build agents failed. Thanks for the quick fix!

When could we expect the fix to be deployed with the .NET SDK?

**Comment by erdembayar** [comment]
@Starkie 
Yes, it would be deployed with the .Net SDK. I'll update here once it's in NET SDK.  Meantime please use older .Net SDK version works for you, thank you for your understanding.

**Comment by erdembayar** [comment]
Here is link for daily build if someone wants to download 5.0.400 daily build
https://github.com/dotnet/installer/blob/main/README.md#installers-and-binaries

**Comment by forderud** [comment]
Just stumbled across this regression after upgrading to VS2019 16.10.1 that ships with NuGet 5.10.0.  This issue is likely breaking workflows for company-internal NuGet servers behind corporate proxy servers. That's at least the case for me.

While waiting for a VS2019 update with a fix, would it be possible to ensure that the issue gets mentioned in the release notes for NuGet 5.10.0?

**Comment by erdembayar** [comment]
I can see fix is already in [16.11.0 preview 2](https://visualstudio.microsoft.com/vs/preview/). But it takes time to go in GA version.
